polyfill/index.d.ts

```diff
@@ -0,0 +1,385 @@
+     * moment.js and Luxon) gives the same result as `earlier` when turning the
+     * clock back, and `later` when setting the clock forward.
+     *
+     * */
+    disambiguation: 'earlier' | 'later' | 'reject';
+    // TODO: should there be a `compatible` (to legacy `Date`, moment, etc.)
+    //       disambiguation option? See #614.
+  };
+
+  /**
+   * Options for arithmetic operations like `plus()` and `minus()`
+   * */
+  export type ArithmeticOptions = {
+    /**
+     * Controls handling of out-of-range arithmetic results.
+     *
+     * If a result is out of range, then `'constrain'` will clamp the result to
+     * the allowed range, while `'reject'` will throw a RangeError.
+     *
+     * The default is `'constrain'`.
+     */
+    disambiguation: 'constrain' | 'reject';
+    // TODO: may change name from `disambiguation` to `overflow`. See #607.
+  };
+
+  /**
+   * Options to control `Duration.prototype.minus()` behavior
+   * */
+  export type DurationMinusOptions = {
+    /**
+     * Controls how to deal with subtractions that result in negative overflows
+     *
+     * In `'balanceConstrain'` mode, negative fields are balanced with the next
+     * highest field so that none of the fields are negative in the result. If
+     * this is not possible, a `RangeError` is thrown.
+     *
+     * In `'balance'` mode, all fields are balanced with the next highest field,
+     * no matter if they are negative or not.
+     *
+     * TODO: add an example of each mode.
+     *
+     * The default is `'balanceConstrain'`.
+     *
+     */
+    disambiguation: 'balanceConstrain' | 'balance';
+  };
+
+  export interface DifferenceOptions<T extends string> {
+    /**
+     * The largest unit to allow in the resulting `Temporal.Duration` object.
+     *
+     * Valid values may include `'years'`, `'months'`, `'days'`, `'hours'`,
+     * `'minutes'`, and `'seconds'`, although some types may throw an exception
+     * if a value is used that would produce an invalid result. For example,
+     * `hours` is not accepted by `Date.prototype.difference()`.
+     *
+     * The default depends on the type being used.
+     *
+     * TODO: document which types use which defaults.
+     */
+    largestUnit: T;
+  }
+
+  export type DurationLike = {
+    years?: number;
+    months?: number;
+    days?: number;
+    hours?: number;
+    minutes?: number;
+    seconds?: number;
+    milliseconds?: number;
+    microseconds?: number;
+    nanoseconds?: number;
+  };
+
+  export class Duration extends Required<DurationLike> {
+    static from(
+      item: Temporal.Duration | DurationLike | string | object,
+      options?: AssignmentOptions
+    ): Temporal.Duration;
+    constructor(
+      years?: number,
+      months?: number,
+      days?: number,
+      hours?: number,
+      minutes?: number,
+      seconds?: number,
+      milliseconds?: number,
+      microseconds?: number,
+      nanoseconds?: number
```

Comment:
Yep, but from TS's perspective default values are part of the implementation, not the type. It makes sense when you think about it: TS is really just a way to create custom lint rules for each JS type. It just tells you what kind of code is legal vs. illegal. So it can tell you that a parameter is required or not, and what kinds of values are allowed or disallowed. But a default parameter value isn't something that would generate a compiler error so TS ignores it. At least that's how I believe it works. 

I actually made the same mistake when I wrote the first draft of the types and had to Google it to find out why `=0` wasn't compiling. ;-) https://stackoverflow.com/questions/35074365/typescript-interface-default-values